macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
 	/// - Parameters:
 	///   - title: String displayed in the button
-	///   - image: The NSImage to display in the button
-	///   - imagePosition: The position of the image, relative to the title
-	///   - style: The ButtonStyle, defaulting to primaryFilled
-	@objc public init(title: String, image: NSImage, imagePosition: NSControl.ImagePosition, style: ButtonStyle = .primaryFilled) {
-		super.init(frame: .zero)
-		self.imagePosition = imagePosition
-		initialize(title: title, image: image, style: style)
+	///   - image: The NSImage to diplay in the button
+	///   - imagePosition: The position of the image relative to the title, default .imageLeading
+	///   - size: The ButtonSize, default .large
+	///   - style: The ButtonStyle, default .primary
+	///   - accentColor: The accent or highlight NSColor, defaulting to the app primary color
+	@objc public convenience init(
+		title: String,
+		image: NSImage,
+		imagePosition: NSControl.ImagePosition = .imageLeading,
+		size: ButtonSize = .large,
+		style: ButtonStyle = .primary,
+		accentColor: NSColor = Colors.primary
+	) {
+		let format = ButtonFormat(size: size, style: style, accentColor: accentColor)
+		self.init(title: title, image: image, imagePosition: imagePosition, format: format)
 	}
 
-	/// Initializes a Fluent UI Button
-	/// Set style to primaryFilled as default
-	@objc public init() {
-		super.init(frame: .zero)
-		initialize(title: nil, image: nil, style: .primaryFilled)
+	/// Initializes a Fluent UI Button with no title or image, and with default formatting
+	@objc public convenience init() {
+		self.init()
 	}
 
 	@available(*, unavailable)
 	required public init?(coder decoder: NSCoder) {
 		preconditionFailure()
 	}
-	
-	private func initialize(title: String?, image: NSImage?, style: ButtonStyle = .primaryFilled) {
-		// Do common initialization work
+
+	/// Swift-only designated initializer accepting ButtonFormat struct.
+	/// - Parameters:
+	///   - title: String displayed in the button, default nil
+	///   - image: The NSImage to diplay in the button, default nil
+	///   - imagePosition: The position of the image, relative to the title, default imageLeading
+	///   - format: The ButtonFormat including size, style and accentColor, with all applicable defaults
+	public init(
+		title: String? = nil,
+		image: NSImage? = nil,
+		imagePosition: NSControl.ImagePosition = .imageLeading,
+		format: ButtonFormat = ButtonFormat()
+	)
+	{
+		super.init(frame: .zero)
 		isBordered = false
 		wantsLayer = true
 		layer?.contentsScale = window?.backingScaleFactor ?? 1.0
 
-		self.style = style
+		if let cell = cell as? ButtonCell {
+			// The Button properties `contentTintColorDisabled`,
+			// `backgroundColorDisabled`, and `borderColorDisabled` fully
+			// specify the look of the disabled button, and may already include
+			// a 'disabled' system effect.  Don't apply the effect again on top
+			// of that.
+			cell.imageDimsWhenDisabled = false
```

Comment:
Would this be true if we weren't using a template image as well?

Reply:
Do the fluent arrows count as not using template images?

Note that the property is not very well named, since the dimming is also applied to the button title.  It's probably updating the contentTintColor.

Reply:
you can test locally it by making /fluentui-apple/macos/FluentUITestApp/Assets.xcassets/ic_fluent_chevron_left_16_filled not render as "template image" in Xcode project 
![Screen Shot 2020-12-07 at 5 53 45 PM](https://user-images.githubusercontent.com/20715435/101428261-371fd080-38b5-11eb-84c0-efee79f84f0e.png)

Reply:
imageDimsWhenDisabled does apply to non-template images.  (contentTintColor does not.)